pkg/scalers/mysql_scaler.go

```diff
@@ -64,101 +63,35 @@ func NewMySQLScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
 }
 
 func parseMySQLMetadata(config *scalersconfig.ScalerConfig) (*mySQLMetadata, error) {
-	meta := mySQLMetadata{}
+	meta := &mySQLMetadata{}
 
-	if val, ok := config.TriggerMetadata["query"]; ok {
-		meta.query = val
-	} else {
-		return nil, fmt.Errorf("no query given")
-	}
-
-	if val, ok := config.TriggerMetadata["queryValue"]; ok {
-		queryValue, err := strconv.ParseFloat(val, 64)
-		if err != nil {
-			return nil, fmt.Errorf("queryValue parsing error %w", err)
-		}
-		meta.queryValue = queryValue
-	} else {
-		if config.AsMetricSource {
-			meta.queryValue = 0
-		} else {
-			return nil, fmt.Errorf("no queryValue given")
-		}
+	if err := config.TypedConfig(meta); err != nil {
+		return nil, fmt.Errorf("error parsing mysql metadata: %w", err)
 	}
 
-	meta.activationQueryValue = 0
-	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
-		activationQueryValue, err := strconv.ParseFloat(val, 64)
-		if err != nil {
-			return nil, fmt.Errorf("activationQueryValue parsing error %w", err)
-		}
-		meta.activationQueryValue = activationQueryValue
-	}
-
-	switch {
-	case config.AuthParams["connectionString"] != "":
-		meta.connectionString = config.AuthParams["connectionString"]
-	case config.TriggerMetadata["connectionStringFromEnv"] != "":
-		meta.connectionString = config.ResolvedEnv[config.TriggerMetadata["connectionStringFromEnv"]]
-	default:
-		meta.connectionString = ""
-		var err error
-		host, err := GetFromAuthOrMeta(config, "host")
-		if err != nil {
-			return nil, err
-		}
-		meta.host = host
-
-		port, err := GetFromAuthOrMeta(config, "port")
-		if err != nil {
-			return nil, err
-		}
-		meta.port = port
-
-		username, err := GetFromAuthOrMeta(config, "username")
-		if err != nil {
-			return nil, err
-		}
-		meta.username = username
-
-		dbName, err := GetFromAuthOrMeta(config, "dbName")
-		if err != nil {
-			return nil, err
-		}
-		meta.dbName = dbName
-
-		if config.AuthParams["password"] != "" {
-			meta.password = config.AuthParams["password"]
-		} else if config.TriggerMetadata["passwordFromEnv"] != "" {
-			meta.password = config.ResolvedEnv[config.TriggerMetadata["passwordFromEnv"]]
-		}
-
-		if len(meta.password) == 0 {
-			return nil, fmt.Errorf("no password given")
-		}
-	}
+	meta.ActivationQueryValue = 0
```

Comment:
this would overwrite any user-defined value since it happens after ` config.TypedConfig(meta)`
```suggestion
```

Reply:
Thanks @wozniakjan . Have addressed it